pageserver/page_api/proto/page_service.proto

```diff
@@ -0,0 +1,212 @@
+// Page service, presented by pageservers for computes.
+//
+// This is the compute read path. It primarily serves page versions at given
+// LSNs, but also base backups, SLRU segments, and relation metadata.
+//
+// EXPERIMENTAL: this is still under development and subject to change.
+//
+// Request metadata headers:
+// - authorization: JWT token ("Bearer <token>"), if auth is enabled
+// - neon-tenant-id: tenant ID ("7c4a1f9e3bd6470c8f3e21a65bd2e980")
+// - neon-shard-id: shard ID, as <number><count> in hex ("0b10" = shard 11 of 16, 0-based)
+// - neon-timeline-id: timeline ID ("f08c4e9a2d5f76b1e3a7c2d8910f4b3e")
+// - neon-compute-mode: compute mode ("primary", "static", "replica")
```

Comment:
A 'replica' can promoted to a 'primary'. What's the implication of that? Do we to start new streams when that happens?

Reply:
I think we should, yeah -- it may not be strictly necessary depending on what we use them for, but it's also a bit of a footgun if we don't.

@VladLazar Do we have a concrete need for these to justify the cost/complexity of re-establishing all streams when promoted?

Reply:
I changed this to a TODO to unblock the PR. We can add it later if we have a compelling need for it.